kasa/aestransport.py

```diff
@@ -207,29 +236,33 @@ async def perform_handshake(self):
 
         _LOGGER.debug(f"Device responded with: {resp_dict}")
 
-        if status_code == 200 and resp_dict["error_code"] == 0:
-            _LOGGER.debug("Decoding handshake key...")
-            handshake_key = resp_dict["result"]["key"]
-
-            self._session_cookie = self._http_client.cookies.get(  # type: ignore
-                self.SESSION_COOKIE_NAME
+        if status_code != 200:
+            raise SmartDeviceException(
+                f"Unable to complete handshake for {self.host}, "
+                + "response status code is {status_code}"
             )
-            if not self._session_cookie:
-                self._session_cookie = self._http_client.cookies.get(  # type: ignore
-                    "SESSIONID"
-                )
 
-            self._session_expire_at = time.time() + 86400
-            self._encryption_session = AesEncyptionSession.create_from_keypair(
-                handshake_key, key_pair
+        self._handle_response_error_code(resp_dict, "Unable to complete handshake")
+
+        _LOGGER.debug("Decoding handshake key...")
```

Comment:
```suggestion
```
Is this too verbose on logging If someone receives an error somewhere below, it's probably easy to trace backwards (or use a debugger) to find the exact position.

Reply:
Removed